### PR TITLE
Add the source data to Point3d objects in Graph3D

### PR DESCRIPTION
**Pull request by TRManderson**
This allows more complex tooltips, using information sent from the server + the source object

### CODE REVIEW

File: lib/graph3d/Graph3d.js [hunk 1]
```diff
@@ -589,6 +590,7 @@ Graph3d.prototype._getDataPoints = function (data) {
       point.x = data[i][this.colX] || 0;
       point.y = data[i][this.colY] || 0;
       point.z = data[i][this.colZ] || 0;
+      point.source = data[i];
```
Comment (mojoaxel) [inline lib/graph3d/Graph3d.js]: About the naming: I personally would prefer `point.data` over `point.source`

**Comment by mojoaxel** [comment]
@TRManderson Could you maybe extend the [graph3d tooltips example](/almende/vis/blob/develop/examples/graph3d/11_tooltips.html) to use the new data? In any case this new feature should be documented and explained [here](/almende/vis/blob/develop/docs/graph3d/index.html). Thanks!

**Comment by TRManderson** [comment]
Updated to include docs and example change. I couldn't come up with a compelling extension to the tooltips example, but I think I demonstrated the functionality non-horribly.

File: examples/graph3d/11_tooltips.html [hunk 2]
```diff
@@ -34,10 +39,10 @@
           var z = custom(x,y);
           if (withValue) {
             var value = (y - x);
-            data.add({x:x, y:y, z: z, style:value});
+            data.add({x:x, y:y, z: z, style:value, extra: extra_content[(x*y) % extra_content.length]});
           }
           else {
-            data.add({x:x, y:y, z: z});
+			data.add({x:x, y:y, z: z, extra: extra_content[(x*y) % extra_content.length]});
```
Comment (mojoaxel) [inline examples/graph3d/11_tooltips.html]: Can you please change the ident every where from "tab" to "spaces". (I also don't like it, but ithe the projects standart.)

**Comment by TRManderson** [comment]
Sorry about that - dodgy vim config meant I was typing tabs when I thought I had spaces. I've rebased develop with the changes requested.

**Comment by mojoaxel** [comment]
@TRManderson Thanks!

**Pull request merged**